src/test/java/io/vertx/core/DeploymentTest.java

```diff
@@ -1234,6 +1234,28 @@ public void testUndeployAll() throws Exception {
     vertx = null;
   }
 
+  @Test
+  public void testDeployChildOnParentUndeploy() throws Exception {
+    class ParentVerticle extends AbstractVerticle {
+      @Override
+      public void stop(Promise<Void> stopPromise) {
+        vertx.deployVerticle(ChildVerticle.class.getName())
+          .<Void>mapEmpty()
+          .setHandler(stopPromise);
+      }
+    }
+
+    CountDownLatch latch = new CountDownLatch(1);
+    vertx.deployVerticle(new ParentVerticle(), ar1 -> {
+      assertTrue(ar1.succeeded());
+      vertx.undeploy(ar1.result(), ar2 -> {
```

Comment:
you can use the helper method onSuccess / onFailure to make the test shorter and easier to read

Reply:
Done